File: LeetCode Solutions/Problems/2335. Minimum Amount of Time to Fill Cups.java
class Solution {
    public int fillCups(int[] amount) {
        int maximum = 0, sum = 0;
        for(int a: amount) {
            maximum = Math.max(a, maximum);
            sum += a;
        }
        return Math.max(maximum, (sum + 1) / 2);
    }
}

File: LeetCode Solutions/Problems/2335. Minimum Amount of Time to Fill Cups.cpp
class Solution {
public:
    int fillCups(vector<int>& amount) {
        int maximum = 0, sum = 0;
        for(int& a: amount) {
            maximum = max(a, maximum);
            sum += a;
        }
        return max(maximum, (sum + 1) / 2);
    }
};